### PR TITLE
SISRP-13257, controller to manage session during delegate-view-as mode

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.berkeley.edu/browse/SISRP-13257

You'll see that `DelegateActAsController` extends `ActAsController`. If shared logic becomes a problem we'll split off the `delegate-act-as` code. After this merge I'll drop in `footer.html` and related.

### CODE REVIEW

File: src/assets/javascripts/angular/factories/adminFactory.js [hunk 1]
```diff
@@ -53,12 +63,14 @@ angular.module('calcentral.factories').factory('adminFactory', function(apiServi
   };
 
   return {
+    delegateActAs: delegateActAs,
```
Comment (christianvuerings) [inline src/assets/javascripts/angular/factories/adminFactory.js]: @johncrossman fyi - usually we try to alphabetize these

File: src/assets/javascripts/angular/controllers/widgets/delegateActingAsController.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * Controller for delegate users in act-as mode.
+ */
+angular.module('calcentral.controllers').controller('DelegateActingAsController', function(adminFactory, apiService, $scope) {
+  $scope.delegate = {};
+
+  /**
+   * Stop acting as someone else
+   */
+  $scope.delegate.stopActAs = function() {
+    adminFactory.stopDelegateActAs().success(apiService.util.redirectToToolbox).error(apiService.util.redirectToToolbox);
```
Comment (christianvuerings) [inline src/assets/javascripts/angular/controllers/widgets/delegateActingAsController.js]: @johncrossman if `success` and `error` is the same, we should probably use `.then` instead.

**Comment by johncrossman** [comment]
Thanks @christianvuerings  I've amended the commit.

File: config/routes.rb [hunk 3]
```diff
@@ -129,6 +129,8 @@
   end
 
   # Act-as endpoints
+  post '/delegate_act_as' => 'delegate_act_as#start', :via => :post
+  post '/stop_delegate_act_as' => 'delegate_act_as#stop', :via => :post
```
Comment (paulkerschen) [inline config/routes.rb]: @johncrossman FYI, all these `:via` options in our routes are redundant holdovers from an older version of Rails. I'll create a backlog ticket to remove them in our leisure time.

**Comment by paulkerschen** [comment]
images

**Comment by johncrossman** [comment]
Yay! Merged! Thanks, @paulkerschen   
gettin-ziggy-with-it

**Pull request merged**